src/core/dbus-execute.c

```diff
@@ -1301,6 +1334,58 @@ int bus_exec_context_set_transient_property(
         if (streq(name, "RootImage"))
                 return bus_set_transient_path(u, name, &c->root_image, message, flags, error);
 
+        if (streq(name, "RootImageOptions")) {
+                _cleanup_(mount_options_free_allp) MountOptions *options = NULL;
+                _cleanup_free_ char *format_str = NULL;
+                const char *mount_options;
+                unsigned int partition_number;
+
+                r = sd_bus_message_enter_container(message, 'a', "(us)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(us)", &partition_number, &mount_options)) > 0) {
+                        _cleanup_free_ char *previous = TAKE_PTR(format_str);
+                        MountOptions *o = NULL;
+
+                        if (asprintf(&format_str, "%s %u:%s", strempty(previous), partition_number, mount_options) < 0)
+                                return log_oom();
```

Comment:
something needs to be done about spaces embedded in mount strings. either refuse them here, or escape them. otherwise we'll generate formatted strings here that parse differently then we mean them

Reply:
good point, added check here and returns error if it has any whitespace character